backends/kv/reader.go

```diff
@@ -148,6 +149,19 @@ func (ki *Iterator) Next() bool {
 		byName[field.Name] = col
 	}
 
+	if specificColumnsRequested {
```

Comment:
Better add a condition to see if the `len(columnNamesToReturn) != len(columns)` to avoid iterating the systemAttribute for no reason